.travis.yml

```diff
@@ -2,33 +2,52 @@ language: android
 jdk: oraclejdk8
 sudo: false
 
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
 cache:
   directories:
-  - "$HOME/.m2"
-
-before_cache:
-- rm -rf $HOME/.m2/repository/io/cucumber
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+    - $HOME/.android/build-cache
 
 android:
   components:
-  - tools
-  - platform-tools
-  - tools
-  - build-tools-26.0.0
-  - android-21
-  - sys-img-armeabi-v7a-android-21
+    - tools
+    - platform-tools
+    - tools # Note that the tools section appears twice on purpose as it’s required to get the newest Android SDK tools.
+
+    - build-tools-27.0.3
 
-before_install:
-  - echo no | android create avd --force -n test -t android-21 --abi armeabi-v7a
-  - emulator -avd test -no-skin -no-audio -no-window &
+    # The SDK version used to compile your project
+    - android-27
+
+    # Additional components
+    - extra-google-google_play_services
+    - extra-google-m2repository
+    - extra-android-m2repository
+    - addon-google_apis-google-26
+
+    # Specify at least one system image,
+    # if you need to run emulator(s) during your tests
+    - sys-img-armeabi-v7a-android-27
 
 install:
-  - mvn -q install --also-make --projects android,picocontainer -DskipTests=true
-    -Dmaven.javadoc.skip=true
+ - ./gradlew test
 
+
+# Emulator Management: Create, Start and Wait
 before_script:
+  - echo no | android create avd --force -n test -t android-27 --abi armeabi-v7a
```

Comment:
This command currently fails because there is no `armeabi-v7a` for android 27. Do you know which is the right abi to use?

You can try this your self btw. Don't forget to change `     - sys-img-armeabi-v7a-android-27 ` above to match.

```
$ echo no | android create avd --force -n test -t android-27 --abi armeabi-v7a
Valid ABIs: no ABIs.
Error: Invalid --abi armeabi-v7a for the selected target.
```

Reply:
Okay. Looks like the android tool is deprecated in favour of the sdk manager. 

Looks like we'll have to use that instead.